SofaKernel/modules/SofaCore/src/sofa/core/objectmodel/Base.h

```diff
@@ -180,6 +180,9 @@ class SOFA_CORE_API Base
                            std::initializer_list<BaseData*> outputs);
     void addOutputsToCallback(const std::string& name, std::initializer_list<BaseData*> outputs);
 
+
+    virtual std::string getPathName() const { return ""; }
```

Comment:
This is ugly, I don't like it much but I can't put a virtual pure here since BaseContext also inherits Base, which means more than 1 final overrider for Node (inherits BaseNode & BaseContext...)